scipy/stats/_unuran/unuran_wrapper.pyx.templ

```diff
@@ -267,7 +267,11 @@ cdef void _pack_distr(unur_distr *distr, dict callbacks) except *:
             unur_distr_discr_set_cdf(distr, discr_cdf_thunk)
 
 
-def _validate_domain(domain):
+def _validate_domain(domain, dist):
+    if domain is None and hasattr(dist, 'support'):
+        # if the distribution has a support method, use it
+        # to get the domain.
+        domain = dist.support()
     if domain is not None:
```

Comment:
Just to confirm: if the attribute support does not exist, no error is raised since it defaults to -inf, inf?

Reply:
> if the attribute support does not exist, no error is raised since it defaults to -inf, inf?

Yes.